src/identity/manager.rs

```diff
@@ -414,6 +415,19 @@ impl SecretManager<CaClient> {
     }
 }
 
+impl SecretManager<CustomSigner> {
+    pub async fn origin(cfg: crate::config::Config) -> Result<Self, anyhow::Error> {
+        let k8s_client = tls::create_k8s_client(cfg.k8s_ca_crt).await;
+        match k8s_client {
+            Ok(client) => {
+                let custom_signer = CustomSigner::new(Some("kubernetes.io/kube-apiserver-client".to_string()), client)?;
```

Comment:
shouldn't this be cfg.signer? instead of apiserver-client

Reply:
my bad, forgot to remove the testing code.